From 3d49ca2ca918467fa9a54d09d4ee6a7caa089688 Mon Sep 17 00:00:00 2001
From: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Date: Tue, 2 Oct 2018 16:02:43 +0300
Subject: [PATCH 4767/5242] MLK-19716-2 sound: asoc: fix quality select switch

commit  51c16110d586f2537f4d8c950f353ca6432b1edc from
https://source.codeaurora.org/external/imx/linux-imx.git

Value read from register was shifted and compared
with a non-shifted val and the behaviour was that
it allways went on the default branch.

Signed-off-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Reviewed-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_micfil.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/sound/soc/fsl/fsl_micfil.c b/sound/soc/fsl/fsl_micfil.c
index e04dd20..672b628 100644
--- a/sound/soc/fsl/fsl_micfil.c
+++ b/sound/soc/fsl/fsl_micfil.c
@@ -745,8 +745,7 @@ static inline int get_pdm_clk(struct fsl_micfil *micfil,
 		    >> MICFIL_CTRL2_CICOSR_SHIFT);
 
 	regmap_read(micfil->regmap, REG_MICFIL_CTRL2, &ctrl2_reg);
-	qsel = ((ctrl2_reg & MICFIL_CTRL2_QSEL_MASK)
-		>> MICFIL_CTRL2_QSEL_SHIFT);
+	qsel = ctrl2_reg & MICFIL_CTRL2_QSEL_MASK;
 
 	switch (qsel) {
 	case MICFIL_HIGH_QUALITY:
-- 
1.7.9.5

